### DISCUSSION TITLE
Support for strain gradient plasticity in MFront and code_aster

### DISCUSSION THREAD

**Discussion by ngrilli** [thelfer/tfel#45] [Q&A] [answered]
Dear MFront Users,

Do you envisage a strategy to implement strain gradient plasticity using MFront and code aster?

What I want to do is to calculate the gradient of the plastic deformation gradient
in the Meric Cailletaud model and use the values to modify the critical resolved shear stress.

It could use the plastic deformation gradient at the current or previous time step
to calculate the gradient of the plastic deformation gradient, that would not matter.

Thank you very much in advance.

Best Regards,
Nicolò

**Comment by thelfer** [comment, marked as answer]
Hi @ngrilli,

This is a very interesting question. We had a meeting with the `code_aster` development team precisely on that matter two month ago.
There won't be an direct way to do what you want in the near future.

However, I would expect that something can be done in the brand new python interface of `code_aster`. This talk may help: https://github.com/thelfer/tfel-doc/blob/master/MFrontUserDays/FifthUserDays/StrainGradientPlasticity-Sanahuja-MFrontUserDays-2019.pdf

Don't hesitate to contact Julien Sanahuja to discuss with him of this solution. Maybe AsterO'dactyle may help.

But this is not exactly what you have in mind and you might get limited by the fact that `code_aster` only supports a limited number of external state variables.

In this case, you could give `mgis.fenics` a try (https://thelfer.github.io/mgis/web/mgis_fenics.html), I am pretty sure we could do something for you in this project given the flexibility of `FEniCS`.

Best,
Thomas
